### DISCUSSION TITLE
include "funding" field in package.json?

### DISCUSSION THREAD

**Discussion by marcjansen** [geostyler/geostyler#2181] [Ideas]
What do you think about the following change:

```diff
diff --git a/package.json b/package.json
index f171abf4..0015ff49 100644
--- a/package.json
+++ b/package.json
@@ -153,5 +153,6 @@
     "ol": ">=6.x",
     "react": ">=16.x",
     "react-dom": ">=16.x"
-  }
+  },
+  "funding": "https://opencollective.com/geostyler"
 }

```

Then when one runs [`npm fund`](https://docs.npmjs.com/cli/v7/commands/npm-fund) this output would be generated:

```
$ npm fund
https://opencollective.com/geostyler
│ └── geostyler@12.0.0
├─┬ https://opencollective.com/babel
│ │ └── @babel/core@7.20.12
│ └── https://opencollective.com/browserslist
[skipped a lot of lines]
```

What are your thoughts? I could add it to all the geostyler repos, just like the README section or the actual sponsor button on GitHub.

**Comment by jansule** [comment]
Does the info only occur when running `npm fund` or does it also occur on other `npm` commands? If it's the former one, this would be ok for me. But I do not want other devs to get spammed on `npm i` or something alike, which can be seen with other repos sometimes.

**Reply by marcjansen** [reply]
AFAIK it'll be shown when one does `npm i`, but only as a summary like here:

```
$ npm i @babel/core

added 54 packages, removed 4 packages, and audited 55 packages in 16s

4 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```

`npm fund` will then print out the actual funding URLs

**Reply by jansule** [reply]
Thanks for the info. Then :+1: from my side

**Comment by marcjansen** [comment]
See the following PRs for the implementation:

* https://github.com/geostyler/geostyler/pull/2182
* https://github.com/geostyler/geostyler-geojson-parser/pull/488
* https://github.com/geostyler/geostyler-geoserver-plugin/pull/113
* https://github.com/geostyler/geostyler-demo/pull/473
* https://github.com/geostyler/geostyler-style/pull/578
* https://github.com/geostyler/geostyler-wfs-parser/pull/508
* https://github.com/geostyler/geostyler-rest/pull/75
* https://github.com/geostyler/geostyler-converter-demo/pull/21
* https://github.com/geostyler/geostyler-mapbox-parser/pull/235
* https://github.com/geostyler/geostyler-sld-parser/pull/803
* https://github.com/geostyler/geostyler-shapefile-parser/pull/325
* https://github.com/geostyler/geostyler-cql-parser/pull/314
* https://github.com/geostyler/geostyler-qgis-parser/pull/472
* https://github.com/geostyler/geostyler-cli/pull/350
* https://github.com/geostyler/geostyler-data/pull/280
* https://github.com/geostyler/geostyler-symcore-parser/pull/2
* https://github.com/geostyler/geostyler-mapfile-parser/pull/249
* https://github.com/geostyler/geostyler-legend/pull/437
* https://github.com/geostyler/geostyler-openlayers-parser/pull/712

**Comment by marcjansen** [comment]
All repos have the field now. Closing.
